compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -172,7 +172,11 @@ LogicalResult AIETargetBackend::serializeExecutable(
   // generated by `aie2xclbin`
   SmallVector<std::string> entryPointNames;
   for (auto exportOp : variantOp.getExportOps()) {
-    entryPointNames.emplace_back(exportOp.getSymName());
+    // The xclbin kernel name, appended with instance name suffix (`:MLIRAIEV1`,
+    // 10 chars) is required by the xclbinutil to have a length smaller or equal
+    // to 64 right now. To have some additional wiggle room for suffix changes,
+    // we use the 48 first characters for the kernel name.
+    entryPointNames.emplace_back(exportOp.getSymName().substr(0, 48));
```

Comment:
I am not sure this will work for all cases, as the way the naming scheme works, we have kernels like matmul_M1_N1_K1
and then another code be matmul_M1_N1_k2 so both might end up being matmul_ML_N1 with this way.

Reply:
Right, but different dispatches have a separate xclbin? This is the internal kernel name within the xclbin.

Reply:
For me, we can also generate a uuid or something from the kernel name, but I thought that might be obscure as it would not contain any information about the type of operation. = maybe overkill, as personally, I think the xclbinutil should allow kernel names with dynamic size and we could check with the (I believe) XRT team on whether this could be changed.

Reply:
Alternatively, we could work with kernel keys like `KERNEL_0`, `KERNEL_1`, etc and then possibly keep a map around somewhere to the SymName

Reply:
so the entrypointName is put in our vmfb artifact, L268 of this file, and then it is used to find the xclbin of a given dispatch at runtime, 
https://github.com/nod-ai/iree-amd-aie/blob/main/runtime/src/iree-amd-aie/driver/xrt/native_executable.cc#L174
That being said, currently we are supporting only one entry point function per executable (per xclbin), so you are right, this shortening should be fine due to that